pxtblocks/fields/field_note.ts

```diff
@@ -330,6 +330,31 @@ export class FieldNote extends Blockly.FieldNumber implements FieldCustom {
 
         Blockly.DropDownDiv.setColour(this.primaryColour, this.borderColour);
         Blockly.DropDownDiv.showPositionedByBlock(this, this.sourceBlock_ as Blockly.BlockSvg, () => this.onHide());
+
+
+        this.htmlInput_.addEventListener("keydown", this.keyHandler);
+    }
+
+    protected keyHandler = (ev: KeyboardEvent) => {
+        const currentFreq = typeof this.value_ === "string" ? parseFloat(this.value_) : this.value_;
+    
+        if (ev.code === "ArrowUp") {
```

Comment:
This can be simplified to reduce duplication. Something like:

```
        if (ev.code === "ArrowUp" || ev.code === "ArrowDown") {
            const {keyAbove, keyBelow} = this.getNeighboringKeys(currentFreq);
            const newKey = ev.code === "ArrowUp" ? keyAbove : keyBelow;
```

Then use `newKey` for everything below without needing two branches with the same code.